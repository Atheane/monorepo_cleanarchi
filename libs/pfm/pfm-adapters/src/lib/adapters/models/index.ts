export { AggregationBank } from './bank/Bank';
export { AggregationAccount, SMoneyBankAccount } from './bankAccount/BankAccount';
export * from './transaction';
