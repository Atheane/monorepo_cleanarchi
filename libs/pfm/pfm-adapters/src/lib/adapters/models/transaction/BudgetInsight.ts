enum EventTypeBudgetInsight {
  ACCOUNT_SYNC = 'ACCOUNT_SYNC',
}

export type EventBudgetInsight = {
  date: Date;
  type: EventTypeBudgetInsight;
  data: BankAccount;
  version: number;
};

export type BankAccount = {
  id: number;
  id_connection: number;
  id_user: number;
  id_source: number;
  id_parent?: number;
  number: string;
  webid?: string;
  original_name: string;
  balance: number;
  coming?: number;
  display: boolean;
  last_update?: Date;
  deleted?: Date;
  disabled?: string;
  iban?: string;
  currency: Currency;
  id_type: number;
  bookmarked: number;
  name: string;
  error?: string;
  usage: AccountUsage;
  ownership: string;
  company_name?: string;
  bic?: string;
  coming_balance?: string;
  formatted_balance?: string;
  type: AccountType;
  information?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  loan?: Loan;
  transactions: TransactionBudgetInsight[];
};

export type TransactionBudgetInsight = {
  id: number;
  id_account: number;
  webid?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  application_date?: Date;
  date: Date;
  value: number;
  gross_value?: string;
  original_wording: string;
  simplified_wording: string;
  stemmed_wording?: string;
  wording?: string;
  id_category?: number;
  state?: string;
  date_scraped: Date;
  rdate: Date;
  vdate?: Date;
  bdate?: Date;
  coming: boolean;
  active: boolean;
  id_cluster?: number;
  comment?: string;
  last_update?: Date;
  deleted?: Date;
  original_value?: number;
  original_gross_value?: number;
  original_currency?: Currency;
  commission?: number;
  commission_currency?: Currency;
  country?: string;
  counterparty?: string;
  card?: string;
  category: Category;
  type: TransactionTypeBudgetInsight;
  new: boolean;
  formatted_value: string;
  documents_count: number;
  nopurge: number;
};

export enum TransactionTypeBudgetInsight {
  TRANSFER = 'transfer',
  ORDER = 'order',
  CHECK = 'check',
  DEPOSIT = 'deposit',
  PAYBACK = 'payback',
  WITHDRAWAL = 'withdrawal',
  LOAD_PAYMENT = 'loan_payment',
  BANK = 'bank',
  CARD = 'card',
  DEFERRED_CARD = 'deferred_card',
  SUMMARY_CARD = 'summary_card',
}

export type Currency = {
  id: string;
  symbol: string;
  prefix: boolean;
  crypto: boolean;
  precision: number;
  marketcap?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  datetime?: Date;
  name: string;
};

export type Category = {
  id: number;
  id_parent_category?: number;
  name: string;
  income?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  color: string;
  id_parent_category_in_menu?: number;
  name_displayed: string;
  refundable: boolean;
  id_user?: number;
  id_logo?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  hidden: boolean;
  accountant_account?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
};

export enum AccountType {
  UNKOWN = 1,
  CHECKING = 2,
  SAVINGS = 3,
  DEPOSIT = 4,
  LOAN = 5,
  MARKET = 6,
  JOINT = 7,
  CARD = 8,
  LIFEINSURANCE = 9,
  PEE = 10,
  PERCO = 11,
  ARTICLE83 = 12,
  RSP = 13,
  PEA = 14,
  CAPITALISATION = 15,
  PERP = 16,
  MADELIN = 17,
  MORTGAGE = 18,
  CONSUMERCREDIT = 19,
  REVOLVINGCREDIT = 20,
  PER = 21,
}

export enum AccountUsage {
  PRIV = 'PRIV',
  ORGA = 'ORGA',
  ASSO = 'ASSO',
}

export type Loan = {
  id: number;
  id_account: number;
  contact_name?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  total_amount: number;
  available_amount?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  used_amount?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  subscription_date?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  maturity_date?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  next_payment_amount: number;
  next_payment_date?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  rate?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  nb_payments_left?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  nb_payments_done?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  nb_payments_total?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  last_payment_amount?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  last_payment_date?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  account_label?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  insurance_label?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  duration?: any /* eslint-disable @typescript-eslint/no-explicit-any */;
  type: string;
};
