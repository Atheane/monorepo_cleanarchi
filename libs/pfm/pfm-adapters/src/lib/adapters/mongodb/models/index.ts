export * from './AccountStatement';
export * from './BudgetInsight';
export * from './P2p';
export * from './User';
