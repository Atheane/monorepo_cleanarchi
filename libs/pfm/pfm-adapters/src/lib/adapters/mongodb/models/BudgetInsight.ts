import { Document, Schema, Connection, Model } from 'mongoose';
import { EventBudgetInsight } from '../../models/transaction/BudgetInsight';

export enum EventType {
  ACCOUNT_SYNC = 'ACCOUNT_SYNC',
}

export type EventDoc = EventBudgetInsight & Document;

export class EventSchema extends Schema {
  constructor() {
    super(
      {
        date: {
          type: String,
          required: true,
        },
        type: {
          type: String,
          required: true,
          index: true,
        },
        data: {
          type: Object,
          required: true,
        },
        version: {
          type: Number,
          required: true,
          index: true,
        },
      },
      {
        versionKey: false,
      },
    );
  }
}

const EventIdentifier = 'budgetinsight_event';

let EventModel;
export const connectEventModel = (connection: Connection): Model<EventDoc> => {
  EventModel = connection.model<EventDoc>(EventIdentifier, new EventSchema());
  return EventModel;
};

export const getEventModel = (): Model<EventDoc> => EventModel;
