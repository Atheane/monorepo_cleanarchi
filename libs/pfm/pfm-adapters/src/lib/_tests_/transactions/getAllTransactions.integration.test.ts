import * as nock from 'nock';
import * as dateMock from 'jest-date-mock';
import { ConfigService } from '@oney/envs';
import { Amount, NonCardTransaction, P2p, TransactionError, TransactionType } from '@oney/pfm-core';
import { Currency, TransactionSource } from '@oney/common-core';
import { afterAll, beforeAll, beforeEach, describe, expect, it } from '@jest/globals';
import * as path from 'path';
import { DomainDependencies } from '../../di/DomainDependencies';
import { getEventModel } from '../../adapters/mongodb/models/BudgetInsight';
import { seedEvents } from '../fixtures/transactions/seedEvents';
import { SMoneyTransactionsSDDNullOriginalAmount } from '../fixtures/transactions/SMoneyTransactionsSDDNullOriginalAmount';
import { testConfiguration } from '../fixtures/config/config';
import { initMongooseConnection } from '../../adapters/services/MongoService';
import { setupKernel } from '../fixtures/config/config.kernel';
import { P2pTransactions } from '../fixtures/transactions/P2pTransactions';
import { TransactionsWithNullAmounts } from '../fixtures/transactions/TransactionsWithNullAmounts';
import { userTestCredentials } from '../fixtures/bankAccounts/userTestCredentials';
import { userWithNoAggregatedAccounts } from '../fixtures/bankAccounts/userWithNotAggregatedAccounts';
import { getAccountStatementModel, getP2pModel } from '../../adapters/mongodb/models';

const nockBack = nock.back;
nockBack.fixtures = path.resolve(`${__dirname}/../fixtures/transactions`);
nockBack.setMode('record');
const envPath = path.resolve(__dirname + '/../env/test.env');

describe('Get getAllTransactions integration testing', () => {
  let dependencies: DomainDependencies;

  beforeAll(async () => {
    await new ConfigService({ localUri: envPath }).loadEnv();
    const dbConnection = await initMongooseConnection(process.env.MONGO_URL);
    const kernel = await setupKernel(testConfiguration, false, dbConnection);
    await kernel.initIdentityLib();
    kernel.initBlobStorage();
    dependencies = kernel.getDependencies();
    nock.enableNetConnect();
  });

  beforeEach(async () => {
    dateMock.clear();
    await getEventModel().deleteMany({});
    await getP2pModel().deleteMany({});
    await getAccountStatementModel().deleteMany({});
    await getEventModel().insertMany(seedEvents);
  });

  it('should return a SCT OUT with the counterparty', async () => {
    const { nockDone } = await nockBack('getAllTransactions.json');
    const { userId, userToken } = userTestCredentials;

    const result = await dependencies.getAllTransactions.execute({
      userToken,
      userId,
    });

    const transaction = result.find(t => t.refId === 'a-9F3q0Ov') as NonCardTransaction;

    expect(transaction.type).toEqual('TRANSFER');
    expect(transaction.counterParty).toEqual({
      id: '992',
      iban: 'FR02xxxxxxxxxxxxxxxx3L43',
      fullname: 'Testvirement',
    });
    nockDone();
  });

  it('should return a SCT IN with the counterparty', async () => {
    const { nockDone } = await nockBack('getAllTransactions.json');
    const { userId, userToken } = userTestCredentials;

    const result = await dependencies.getAllTransactions.execute({
      userToken,
      userId,
    });

    const transaction = result.find(t => t.refId === 'wZRl5OM2y') as NonCardTransaction;

    expect(transaction.type).toEqual('TRANSFER');
    expect(transaction.counterParty).toEqual({
      id: '992',
      iban: 'FR8112869000020PC0000012K39',
      fullname: 'Corinne Berthier',
    });
    nockDone();
  });

  it('should get all transactions for a user', async () => {
    // need to mock http call here, because we need to get users accountIds
    const { nockDone } = await nockBack('getAllTransactions.json');
    const { userId, userToken } = userTestCredentials;

    const result = await dependencies.getAllTransactions.execute({
      userToken,
      userId,
    });

    const transaction = result.find(t => t.refId === '126348');

    expect(result.length).toEqual(228);
    const { amount } = await dependencies.mappers.transactionMapper.fromDomain(transaction);
    expect(amount).toEqual(76.61);
    expect(transaction.label).toEqual("HALL'S BEER");
    expect(transaction.type).toEqual('CARD');
    nockDone();
  });

  it('should return a quasi cash transaction as a card transaction', async () => {
    const { nockDone } = await nockBack('getAllTransactions.json');
    const { userId, userToken } = userTestCredentials;

    const result = await dependencies.getAllTransactions.execute({
      userToken,
      userId,
    });

    const transaction = result.find(t => t.refId === 'RimG1b3Tf');

    expect(transaction.type).toEqual('CARD');
    nockDone();
  });

  it('should get no transactions for an account which has no SMO operations, no aggregated accounts and no P2P transactions', async () => {
    const { nockDone } = await nockBack('getAllTransactionsNotAggregated.json');
    const result = await dependencies.getAllTransactions.execute(userWithNoAggregatedAccounts);
    expect(result.length).toEqual(0);
    nockDone();
  });

  it('should throw an query validation error if not sortBy date', async () => {
    const { nockDone } = await nockBack('getAllTransactions.json');
    const { userId, userToken } = userTestCredentials;
    const result = dependencies.getAllTransactions.execute({
      userToken,
      userId,
      query: {
        sortBy: 'nimp',
        dateFrom: 1,
        dateTo: 2,
      },
    });
    await expect(result).rejects.toThrow(TransactionError.InvalidQuery);
    nockDone();
  });

  it('should throw an query validation error if wrong enum for sort by date', async () => {
    const { nockDone } = await nockBack('getAllTransactions.json');
    const { userId, userToken } = userTestCredentials;
    const result = dependencies.getAllTransactions.execute({
      userToken,
      userId,
      query: {
        sortBy: 'date;nimp',
        dateFrom: 1,
        dateTo: 2,
      },
    });
    await expect(result).rejects.toThrow(TransactionError.InvalidQuery);
    nockDone();
  });

  it('should throw an query validation error if dateFrom later than dateTo', async () => {
    const { nockDone } = await nockBack('getAllTransactions.json');
    const { userId, userToken } = userTestCredentials;
    const result = dependencies.getAllTransactions.execute({
      userToken,
      userId,
      query: {
        sortBy: 'date;ASC',
        dateFrom: 3,
        dateTo: 2,
      },
    });
    await expect(result).rejects.toThrow(TransactionError.InvalidQuery);
    nockDone();
  });

  it('should throw an query validation error if dateFrom earlier than 2019-01-01', async () => {
    const { nockDone } = await nockBack('getAllTransactions.json');
    const { userId, userToken } = userTestCredentials;
    const result = dependencies.getAllTransactions.execute({
      userToken,
      userId,
      query: {
        sortBy: 'date;ASC',
        dateFrom: 3,
        dateTo: 5,
      },
    });
    await expect(result).rejects.toThrow(TransactionError.InvalidQuery);
    nockDone();
  });

  it('should get transaction for a specific date in ASC order', async () => {
    const { nockDone } = await nockBack('getTransactionsByDate.json');
    const { userId, userToken } = userTestCredentials;
    const result = await dependencies.getAllTransactions.execute({
      userToken,
      userId,
      query: {
        sortBy: 'date;ASC',
        dateFrom: new Date('2020-09-01').getTime(),
        dateTo: new Date('2020-09-05').getTime(),
      },
    });

    expect(result.length).toEqual(12);
    expect(
      result.every(
        t =>
          new Date(t.date).getTime() >= new Date('2020-09-01').getTime() &&
          new Date(t.date).getTime() < new Date('2020-09-05').getTime(),
      ),
    ).toBeTruthy();
    expect(new Date(result[0].date).getTime()).toBeLessThanOrEqual(new Date(result[1].date).getTime());
    nockDone();
  });

  it('should get p2p among other transactions for all accounts', async () => {
    dateMock.advanceTo(new Date('2020-09-04T16:32:29.047Z'));
    const { nockDone } = await nockBack('getTransactionsByDate.json');
    const { userId, userToken } = userTestCredentials;
    await Promise.all(P2pTransactions.map(p2p => dependencies.p2pRepository.create(new P2p(p2p))));

    const result = await dependencies.getAllTransactions.execute({
      userToken,
      userId,
      query: {
        sortBy: 'date;ASC',
        dateFrom: new Date('2020-09-01').getTime(),
        dateTo: new Date('2020-09-05').getTime(),
      },
    });

    expect(result).toMatchSnapshot();
    nockDone();
  });

  it('should not get p2p among other transactions for all accounts when date not corresponding', async () => {
    dateMock.advanceTo(new Date('2020-08-01T16:32:29.047Z'));
    const { nockDone } = await nockBack('getTransactionsByDate.json');
    const { userId, userToken } = userTestCredentials;
    await Promise.all(P2pTransactions.map(p2p => dependencies.p2pRepository.create(new P2p(p2p))));

    const result = await dependencies.getAllTransactions.execute({
      userToken,
      userId,
      query: {
        sortBy: 'date;ASC',
        dateFrom: new Date('2020-09-01').getTime(),
        dateTo: new Date('2020-09-05').getTime(),
      },
    });

    expect(result).toMatchSnapshot();
    nockDone();
  });

  it('should get SMO SDD transactions and other special cases covered by mapper', async () => {
    const account = {
      id: '1388',
      name: 'Corinne Berthier',
      number: null,
      currency: Currency.EUR,
      balance: new Amount(4210.54),
      metadatas: {
        iban: 'FR8112869000020PC0000012K39',
        fullname: 'John Doe',
      },
      bank: {
        name: 'Oney Banque Digitale',
        logo: 'https://logo.clearbit.com/oney.fr?size=200',
        source: TransactionSource.ODB,
      },
    };
    const result = await SMoneyTransactionsSDDNullOriginalAmount.map(raw =>
      dependencies.mappers.transactionSMoneyMapper.toDomain({
        accounts: [account],
        raw,
      }),
    );
    expect(result[0].type).toEqual(TransactionType.ORDER);
  });

  it('should allow for negative balance', async () => {
    const account = {
      id: '1388',
      name: 'Corinne Berthier',
      number: null,
      currency: Currency.EUR,
      balance: new Amount(-4210.54),
      metadatas: {
        iban: 'FR8112869000020PC0000012K39',
        fullname: 'John Doe',
      },
      bank: {
        name: 'Oney Banque Digitale',
        logo: 'https://logo.clearbit.com/oney.fr?size=200',
        source: TransactionSource.ODB,
      },
    };
    expect(account.balance.value).toEqual(-4210.54);
  });

  it('should give an empty array if error with transaction services', async () => {
    const { nockDone } = await nockBack('getEmptyTransactionsArray.json');
    const result = await dependencies.getTransactionService.getAllTransactions({
      accountsIds: [],
      userId: 'unknown',
      userToken: 'tchatche',
    });
    expect(result).toEqual([]);
    nockDone();
  });

  it('should give an empty array if no transactions', async () => {
    const { nockDone } = await nockBack('getEmptyTransactionsArray.json');
    const { userId, userToken } = userWithNoAggregatedAccounts;
    const result = await dependencies.getTransactionService.getAllTransactions({
      accountsIds: [],
      userId,
      userToken,
      transactionSources: [TransactionSource.ODB],
    });
    expect(result).toEqual([]);
    nockDone();
  });

  it('should return null if amount, originalAMount, currency or fees are null', async () => {
    const formattedTransactions = TransactionsWithNullAmounts.map(t =>
      dependencies.mappers.transactionMapper.fromDomain(t),
    );
    expect(formattedTransactions[0].originalAmount).toBeNull();
    expect(formattedTransactions[0].amount).toEqual(20);
    expect(formattedTransactions[0].fees).toEqual(20);
  });

  afterAll(async () => {
    nock.enableNetConnect();
  });
});
