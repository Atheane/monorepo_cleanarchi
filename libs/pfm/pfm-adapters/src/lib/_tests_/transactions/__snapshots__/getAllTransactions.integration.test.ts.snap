// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get getAllTransactions integration testing should get p2p among other transactions for all accounts 1`] = `
Array [
  CardTransaction {
    "amount": AmountPositive {
      "amount": 76.61,
    },
    "bankAccountId": "8254",
    "card": Object {
      "cardId": null,
      "merchant": null,
      "pan": null,
    },
    "conversionRate": null,
    "currency": "EUR",
    "date": "2020-09-01",
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "HALL'S BEER",
    "originalAmount": AmountPositive {
      "amount": 64.92,
    },
    "refId": "123456",
    "rejectionReason": null,
    "status": "CLEARED",
    "type": "CARD",
  },
  CardTransaction {
    "amount": AmountPositive {
      "amount": 17.93,
    },
    "bankAccountId": "7866",
    "card": Object {
      "cardId": null,
      "merchant": null,
      "pan": null,
    },
    "conversionRate": null,
    "currency": null,
    "date": "2020-09-01",
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "BIECENTRAL LEVIS",
    "originalAmount": null,
    "refId": "126797",
    "rejectionReason": null,
    "status": "CLEARED",
    "type": "CARD",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 1.8,
    },
    "bankAccountId": "7866",
    "conversionRate": null,
    "currency": null,
    "date": "2020-09-01",
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "ABONNEMENT VOTRE FORMULE ZEN",
    "originalAmount": null,
    "refId": "126798",
    "rejectionReason": null,
    "status": "CLEARED",
    "type": "BANK",
  },
  CardTransaction {
    "amount": AmountPositive {
      "amount": 64.92,
    },
    "bankAccountId": "8253",
    "card": Object {
      "cardId": null,
      "merchant": null,
      "pan": null,
    },
    "conversionRate": null,
    "currency": null,
    "date": "2020-09-01",
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "HALL'S BEER",
    "originalAmount": null,
    "refId": "126296",
    "rejectionReason": null,
    "status": "CLEARED",
    "type": "CARD",
  },
  CardTransaction {
    "amount": AmountPositive {
      "amount": 76.61,
    },
    "bankAccountId": "8254",
    "card": Object {
      "cardId": null,
      "merchant": null,
      "pan": null,
    },
    "conversionRate": null,
    "currency": "EUR",
    "date": "2020-09-01",
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "HALL'S BEER",
    "originalAmount": AmountPositive {
      "amount": 64.92,
    },
    "refId": "126348",
    "rejectionReason": null,
    "status": "CLEARED",
    "type": "CARD",
  },
  CardTransaction {
    "amount": AmountPositive {
      "amount": 64.92,
    },
    "bankAccountId": "8253",
    "card": Object {
      "cardId": null,
      "merchant": null,
      "pan": null,
    },
    "conversionRate": null,
    "currency": null,
    "date": "2020-09-01",
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "HALL'S BEER",
    "originalAmount": null,
    "refId": "126347",
    "rejectionReason": null,
    "status": "PENDING",
    "type": "CARD",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 2,
    },
    "bankAccountId": "1388",
    "clearingDate": null,
    "conversionRate": null,
    "counterParty": Object {
      "fullname": "Testvirement",
      "iban": "FR02xxxxxxxxxxxxxxxx3L43",
      "id": "992",
    },
    "currency": "EUR",
    "date": 2020-09-01T17:06:03.000Z,
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "Not provided",
    "originalAmount": AmountPositive {
      "amount": 2,
    },
    "refId": "a-9F3q0Ov",
    "rejectionReason": null,
    "status": "CLEARED",
    "type": "TRANSFER",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 2,
    },
    "bankAccountId": "1388",
    "clearingDate": null,
    "conversionRate": null,
    "counterParty": Object {
      "fullname": "Testvirement",
      "iban": "FR02xxxxxxxxxxxxxxxx3L43",
      "id": "992",
    },
    "currency": "EUR",
    "date": 2020-09-01T17:06:16.000Z,
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "Not provided",
    "originalAmount": AmountPositive {
      "amount": 2,
    },
    "refId": "wZRl5OM2y",
    "rejectionReason": null,
    "status": "CLEARED",
    "type": "TRANSFER",
  },
  CardTransaction {
    "amount": AmountPositive {
      "amount": 120.4,
    },
    "bankAccountId": "8241",
    "card": Object {
      "cardId": null,
      "merchant": null,
      "pan": null,
    },
    "conversionRate": null,
    "currency": null,
    "date": "2020-09-03",
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "Op√©ration Carte",
    "originalAmount": null,
    "refId": "126796",
    "rejectionReason": null,
    "status": "CLEARED",
    "type": "CARD",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 1000,
    },
    "bankAccountId": "1388",
    "clearingDate": 2020-09-03T08:23:13.512Z,
    "conversionRate": null,
    "counterParty": Object {
      "fullname": "Jane Doe",
      "iban": "iban_value",
      "id": "456",
    },
    "currency": "EUR",
    "date": 2020-09-03T08:23:13.512Z,
    "direction": "IN",
    "fees": null,
    "isDeposit": false,
    "label": "p2p received",
    "originalAmount": AmountPositive {
      "amount": 1000,
    },
    "refId": "gaerh",
    "rejectionReason": null,
    "status": "CLEARED",
    "tag": "tag_value",
    "type": "P2P",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 4,
    },
    "bankAccountId": "1388",
    "clearingDate": 2020-09-03T16:20:45.000Z,
    "conversionRate": 1,
    "currency": undefined,
    "date": 2020-09-03T16:20:45.000Z,
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "BNP Paribas",
    "originalAmount": AmountPositive {
      "amount": 4,
    },
    "refId": "Tdle-sdyn",
    "rejectionReason": null,
    "status": "PENDING",
    "type": "ATM",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 4,
    },
    "bankAccountId": "1388",
    "clearingDate": 2020-09-03T16:21:07.000Z,
    "conversionRate": 1,
    "currency": undefined,
    "date": 2020-09-03T16:21:07.000Z,
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "BNP Paribas",
    "originalAmount": AmountPositive {
      "amount": 4,
    },
    "refId": "vAPKLJrJP",
    "rejectionReason": null,
    "status": "PENDING",
    "type": "ATM",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 6,
    },
    "bankAccountId": "1388",
    "clearingDate": 2020-09-03T16:24:42.000Z,
    "conversionRate": 1,
    "currency": undefined,
    "date": 2020-09-03T16:24:42.000Z,
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "BNP Paribas",
    "originalAmount": AmountPositive {
      "amount": 6,
    },
    "refId": "DSPxRpsta",
    "rejectionReason": null,
    "status": "PENDING",
    "type": "ATM",
  },
]
`;

exports[`Get getAllTransactions integration testing should not get p2p among other transactions for all accounts when date not corresponding 1`] = `
Array [
  CardTransaction {
    "amount": AmountPositive {
      "amount": 76.61,
    },
    "bankAccountId": "8254",
    "card": Object {
      "cardId": null,
      "merchant": null,
      "pan": null,
    },
    "conversionRate": null,
    "currency": "EUR",
    "date": "2020-09-01",
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "HALL'S BEER",
    "originalAmount": AmountPositive {
      "amount": 64.92,
    },
    "refId": "123456",
    "rejectionReason": null,
    "status": "CLEARED",
    "type": "CARD",
  },
  CardTransaction {
    "amount": AmountPositive {
      "amount": 17.93,
    },
    "bankAccountId": "7866",
    "card": Object {
      "cardId": null,
      "merchant": null,
      "pan": null,
    },
    "conversionRate": null,
    "currency": null,
    "date": "2020-09-01",
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "BIECENTRAL LEVIS",
    "originalAmount": null,
    "refId": "126797",
    "rejectionReason": null,
    "status": "CLEARED",
    "type": "CARD",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 1.8,
    },
    "bankAccountId": "7866",
    "conversionRate": null,
    "currency": null,
    "date": "2020-09-01",
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "ABONNEMENT VOTRE FORMULE ZEN",
    "originalAmount": null,
    "refId": "126798",
    "rejectionReason": null,
    "status": "CLEARED",
    "type": "BANK",
  },
  CardTransaction {
    "amount": AmountPositive {
      "amount": 64.92,
    },
    "bankAccountId": "8253",
    "card": Object {
      "cardId": null,
      "merchant": null,
      "pan": null,
    },
    "conversionRate": null,
    "currency": null,
    "date": "2020-09-01",
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "HALL'S BEER",
    "originalAmount": null,
    "refId": "126296",
    "rejectionReason": null,
    "status": "CLEARED",
    "type": "CARD",
  },
  CardTransaction {
    "amount": AmountPositive {
      "amount": 76.61,
    },
    "bankAccountId": "8254",
    "card": Object {
      "cardId": null,
      "merchant": null,
      "pan": null,
    },
    "conversionRate": null,
    "currency": "EUR",
    "date": "2020-09-01",
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "HALL'S BEER",
    "originalAmount": AmountPositive {
      "amount": 64.92,
    },
    "refId": "126348",
    "rejectionReason": null,
    "status": "CLEARED",
    "type": "CARD",
  },
  CardTransaction {
    "amount": AmountPositive {
      "amount": 64.92,
    },
    "bankAccountId": "8253",
    "card": Object {
      "cardId": null,
      "merchant": null,
      "pan": null,
    },
    "conversionRate": null,
    "currency": null,
    "date": "2020-09-01",
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "HALL'S BEER",
    "originalAmount": null,
    "refId": "126347",
    "rejectionReason": null,
    "status": "PENDING",
    "type": "CARD",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 2,
    },
    "bankAccountId": "1388",
    "clearingDate": null,
    "conversionRate": null,
    "counterParty": Object {
      "fullname": "Testvirement",
      "iban": "FR02xxxxxxxxxxxxxxxx3L43",
      "id": "992",
    },
    "currency": "EUR",
    "date": 2020-09-01T17:06:03.000Z,
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "Not provided",
    "originalAmount": AmountPositive {
      "amount": 2,
    },
    "refId": "a-9F3q0Ov",
    "rejectionReason": null,
    "status": "CLEARED",
    "type": "TRANSFER",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 2,
    },
    "bankAccountId": "1388",
    "clearingDate": null,
    "conversionRate": null,
    "counterParty": Object {
      "fullname": "Testvirement",
      "iban": "FR02xxxxxxxxxxxxxxxx3L43",
      "id": "992",
    },
    "currency": "EUR",
    "date": 2020-09-01T17:06:16.000Z,
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "Not provided",
    "originalAmount": AmountPositive {
      "amount": 2,
    },
    "refId": "wZRl5OM2y",
    "rejectionReason": null,
    "status": "CLEARED",
    "type": "TRANSFER",
  },
  CardTransaction {
    "amount": AmountPositive {
      "amount": 120.4,
    },
    "bankAccountId": "8241",
    "card": Object {
      "cardId": null,
      "merchant": null,
      "pan": null,
    },
    "conversionRate": null,
    "currency": null,
    "date": "2020-09-03",
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "Op√©ration Carte",
    "originalAmount": null,
    "refId": "126796",
    "rejectionReason": null,
    "status": "CLEARED",
    "type": "CARD",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 1000,
    },
    "bankAccountId": "1388",
    "clearingDate": 2020-09-03T08:23:13.512Z,
    "conversionRate": null,
    "counterParty": Object {
      "fullname": "Jane Doe",
      "iban": "iban_value",
      "id": "456",
    },
    "currency": "EUR",
    "date": 2020-09-03T08:23:13.512Z,
    "direction": "IN",
    "fees": null,
    "isDeposit": false,
    "label": "p2p received",
    "originalAmount": AmountPositive {
      "amount": 1000,
    },
    "refId": "gaerh",
    "rejectionReason": null,
    "status": "CLEARED",
    "tag": "tag_value",
    "type": "P2P",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 4,
    },
    "bankAccountId": "1388",
    "clearingDate": 2020-09-03T16:20:45.000Z,
    "conversionRate": 1,
    "currency": undefined,
    "date": 2020-09-03T16:20:45.000Z,
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "BNP Paribas",
    "originalAmount": AmountPositive {
      "amount": 4,
    },
    "refId": "Tdle-sdyn",
    "rejectionReason": null,
    "status": "PENDING",
    "type": "ATM",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 4,
    },
    "bankAccountId": "1388",
    "clearingDate": 2020-09-03T16:21:07.000Z,
    "conversionRate": 1,
    "currency": undefined,
    "date": 2020-09-03T16:21:07.000Z,
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "BNP Paribas",
    "originalAmount": AmountPositive {
      "amount": 4,
    },
    "refId": "vAPKLJrJP",
    "rejectionReason": null,
    "status": "PENDING",
    "type": "ATM",
  },
  NonCardTransaction {
    "amount": AmountPositive {
      "amount": 6,
    },
    "bankAccountId": "1388",
    "clearingDate": 2020-09-03T16:24:42.000Z,
    "conversionRate": 1,
    "currency": undefined,
    "date": 2020-09-03T16:24:42.000Z,
    "direction": "OUT",
    "fees": null,
    "isDeposit": false,
    "label": "BNP Paribas",
    "originalAmount": AmountPositive {
      "amount": 6,
    },
    "refId": "DSPxRpsta",
    "rejectionReason": null,
    "status": "PENDING",
    "type": "ATM",
  },
]
`;
